waku/v2/protocol/waku_peer_exchange/protocol.nim

```diff
@@ -49,70 +49,53 @@ type
   WakuPeerExchange* = ref object of LPProtocol
     peerManager*: PeerManager
     wakuDiscv5: Option[WakuDiscoveryV5]
-    enrCache: seq[enr.Record] # todo: next step: ring buffer; future: implement cache satisfying https://rfc.vac.dev/spec/34/
+    enrCache*: seq[enr.Record] # todo: next step: ring buffer; future: implement cache satisfying https://rfc.vac.dev/spec/34/
 
-proc sendPeerExchangeRpcToPeer(wpx: WakuPeerExchange, rpc: PeerExchangeRpc, peer: RemotePeerInfo | PeerId): Future[WakuPeerExchangeResult[void]] {.async, gcsafe.} =
+proc request*(wpx: WakuPeerExchange, numPeers: uint64, conn: Connection): Future[WakuPeerExchangeResult[PeerExchangeResponse]] {.async, gcsafe.} =
```

Comment:
Why not split this (and `respond*`) into a `client` module?

Reply:
mm, since the px protocol is quite simple, i would prefer to keep everything in the same file if its fine. easier to navigate.

Reply:
Honestly, I am with @jm-clius. 

Keeping everything in the same module opens the doors for the code to be modified in a way that makes it convoluted and, in the end, non-extensible and unmaintainable.